#Common
common_params:
  verbosity: 0                         #Debug Output, 0-disable, higher for more verbose output
  scanPeriod: 0.1                          #Expected scan period for input pointcloud in seconds. Used of Distortion correction

#Factor Graph
graph_params:
  lidarRate: 10
  imuRate: 100                             #Rate of IMU input (Hz)
  imuTimeOffset: 0.0 #seconds              #Offset between IMU and LiDAR Measurements - Depending on LiDAR timestamp first(+0.05) or last(-0.05)
  smootherLag: 6.0                         #Lag of fixed lag smoother[seconds], not needed for ISAM2
  additonalIterations: 1                  #Additional iterations of graph optimizer after update with new factors
  findUnusedFactorSlots: false
  enableDetailedResults: false
## Noise Parameters
noise_params:
  ### IMU
  #### Accelerometer
  accNoiseDensity: 7.225e-07                 #Continuous-time "Covariance" of accelerometer, microstrain: sigma^2=7.225e-7
  accBiasRandomWalk: 1.0e-06               #Continuous-time "Covariance" describing accelerometer bias random walk, default: 1.0e-06
  accBiasPrior: 0.0                        #Prior/starting value of accelerometer bias
  #### Gyro
  gyrNoiseDensity: 1.71e-08               #Continuous-time "Covariance" of gyroscope measurements, microstrain: sigma^2=1.71-08
  gyrBiasRandomWalk: 9.33e-08              #Continuous-time "Covariance" describing gyroscope bias random walk, default: 9.33e-08
  gyrBiasPrior: 0.0                        #Prior/starting value of gyroscope bias
  #### Preintegration
  integrationNoiseDensity: 1.0         #continuous-time "Covariance" describing integration uncertainty, default: 1.0e-06
  biasAccOmegaPreint: 1.0                 #covariance of bias used for preintegration, default: 1.0e-2
  ### LiDAR
  poseBetweenNoise: [2.0e-02, 2.0e-02, 2.0e-02, 5.0e-01, 5.0e-01, 5.0e-01] #Noise of add between factor -ORDER RPY(rad) - XYZ(meters)
  ### GNSS
  gnssUnaryNoise: 1.0 #1.0e-02
https://images:
  samsung:
    com/de/smartphones/galaxy-z-flip3-5g/watches/watchstudio/SM-R870NZK_ET-SFR86SG_S:
      png?$ORIGIN_PNG$:# Relinearization
relinearization_params:
  positionReLinTh: 5.0e-03                    #Position linearization threshold
  rotationReLinTh: 1.0e-03                    #Rotation linearization threshold
  velocityReLinTh: 1.0e-03                     #Linear Velocity linearization threshold
  accBiasReLinTh: 1.0e-04                      #Accelerometer bias linearization threshold
  gyrBiasReLinTh: 1.0e-04                      #Gyroscope bias linearization threshold
  relinearizeSkip: 1
  enableRelinearization: true
  evaluateNonlinearError: false
  cacheLinearizedFactors: true
  enablePartialRelinearizationCheck: true